### PR TITLE
[Cherrypick][Port] Cultural Invariant Conversion of datatypes during mutation

### PR DESCRIPTION
**Pull request by seantleonard**
## Port Metadata
- porting from `main` to `release/1.2`
  - Issue: #2284 
  - PR: #2307

## Why make this change?

It addresses https://github.com/Azure/data-api-builder/issues/2284 bug raised by the customer.

~~**Note:** Made change in`src/Core/Resolvers/SqlMutationEngine.cs` just for consistency.~~ [SL: this was in original PR, but is not accurate, since no changes to SqlMutationEngine.cs are included

## What is this change?
Made sure datatype conversion is `CultureInfo.InvariantCulture`

### CODE REVIEW

File: src/Core/Services/TypeHelper.cs [hunk 1]
```diff
@@ -304,12 +305,12 @@ public static bool TryGetDbTypeFromSqlDbDateTimeType(SqlDbType sqlDbType, [NotNu
...
             return valueKind switch
             {
-                SyntaxKind.IntValue => Convert.ToInt32(node.Value), // spec
-                SyntaxKind.FloatValue => Convert.ToDouble(node.Value), // spec
+                SyntaxKind.IntValue => Convert.ToInt32(node.Value, CultureInfo.InvariantCulture), // spec
+                SyntaxKind.FloatValue => Convert.ToDouble(node.Value, CultureInfo.InvariantCulture), // spec
                 SyntaxKind.BooleanValue => Convert.ToBoolean(node.Value), // spec
-                SyntaxKind.StringValue => Convert.ToString(node.Value), // spec
+                SyntaxKind.StringValue => Convert.ToString(node.Value, CultureInfo.InvariantCulture), // spec
                 SyntaxKind.NullValue => null, // spec
```
Comment (Aniruddh25) [inline src/Core/Services/TypeHelper.cs:312]: not for this PR, but do you know what the comment `// spec` refers to here?

File: src/Directory.Build.props [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <Nullable>enable</Nullable>
     <BaseOutputPath>..\out</BaseOutputPath>
-    <Version>1.1</Version>
```
Comment (Aniruddh25) [inline src/Directory.Build.props:5]: Is this part of this PR?

**Review by Aniruddh25 — APPROVED** [review]
Looks good.

File: src/Directory.Build.props [hunk 3]
```diff
@@ -2,7 +2,7 @@
...
+    <Version>1.2</Version>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (Aniruddh25) [inline src/Directory.Build.props:8]: The PR description still says: there was a change made in SqlMutationEngine.cs but looking at the original PR, those changed were removed. Looks like the original PR didnt update the description, lets at least update in this cherry-pick PR. 
Also, do we have a similar issue for SQL? If yes, we should fix that too in this version 1.2 (in a separate PR)

File: src/Directory.Build.props (hunk 2 above)
Comment (seantleonard) [reply src/Directory.Build.props:5]: Investigating. Looks like i added 2 commits.

File: src/Service.Tests/CosmosTests/TestBase.cs [hunk 4]
```diff
@@ -46,7 +46,7 @@ public class TestBase
     id : ID!,
     name : String,
     character: Character,
-    age : Int,
+    age : Float,
```
Comment (abhishekkumams) [inline src/Service.Tests/CosmosTests/TestBase.cs:49]: why age is not Int?

**Review by abhishekkumams — APPROVED** [review]

File: src/Directory.Build.props (hunk 2 above)
Comment (seantleonard) [reply src/Directory.Build.props]: I only added one commit, i just need to mark the "merge to" branch as `release/1.2` in GitHub. Fixed.

File: src/Directory.Build.props (hunk 3 above)
Comment (seantleonard) [reply src/Directory.Build.props]: 1. Updated this PR's description to highlight the fact no changes to sqlmutationengine are included.
2. The fix from the original PR affects "typehelper.cs" which is universal. I added a work item to add mirroring tests for SQL: #2319

File: src/Service.Tests/CosmosTests/TestBase.cs (hunk 4 above)
Comment (seantleonard) [reply src/Service.Tests/CosmosTests/TestBase.cs:49]: That was Sourabh's choice to make in #2307 to test this behavior. When we make similar test for SQL via #2319 we don't have to go that route.

File: src/Core/Services/TypeHelper.cs (hunk 1 above)
Comment (seantleonard) [reply src/Core/Services/TypeHelper.cs:312]: I don't recall. That was something Michael added in his perf improvement PR. 'spec' probably aligns with hotchocolate internals is my best guess.

**Pull request merged**